drivers/i2c/tegra_i2c.c

```diff
@@ -345,20 +395,45 @@ static int tegra_i2c_probe(struct udevice *dev)
 	 * We don't have a binding for pinmux yet. Leave it out for now. So
 	 * far no one needs anything other than the default.
 	 */
+#ifndef CONFIG_TEGRA186
 	i2c_bus->pinmux_config = FUNCMUX_DEFAULT;
-	i2c_bus->periph_id = clock_decode_periph_id(blob, node);
+#endif
 
-	/*
-	 * We can't specify the pinmux config in the fdt, so I2C2 will not
-	 * work on Seaboard. It normally has no devices on it anyway.
-	 * You could add in this little hack if you need to use it.
-	 * The correct solution is a pinmux binding in the fdt.
-	 *
-	 *	if (i2c_bus->periph_id == PERIPH_ID_I2C2)
-	 *		i2c_bus->pinmux_config = FUNCMUX_I2C2_PTA;
-	 */
-	if (i2c_bus->periph_id == -1)
-		return -EINVAL;
+#if defined(CONFIG_CLK) || defined(CONFIG_DM_RESET)
+	{
+		int ret;
+		ret = reset_get_by_name(dev, "i2c", &i2c_bus->reset_ctl);
+		if (ret) {
+			debug("reset_get_by_name() failed: %d\n", ret);
```

Comment:
Let's make these messages error() rather than debug() so that people know if/why the driver doesn't probe() correctly. I will fix the SD/MMC driver likewise.